### PR TITLE
Updates regexp-match-indices feature to support 'd' flag

### PR DESCRIPTION
**Pull request by rbuckton**
This brings engine262 up-to-date with the changes for `regexp-match-indices` to use a `d` flag to opt in to returning `.indices` when calling `.exec()` (or any other function that calls `RegExpBuiltinExec`.

### CODE REVIEW

**Comment by rbuckton** [comment]
Note: this was tested against https://github.com/tc39/test262/pull/2934

**Comment by devsnek** [comment]
Thank you very much for this. When the tests land I will merge this :+1:

File: src/parser/Lexer.mjs [hunk 1]
```diff
@@ -887,7 +888,7 @@ export class Lexer {
       }
       const c = this.source[this.position];
       if (isRegularExpressionFlagPart(c)
-          && 'gimsuy'.includes(c)
+          && (surroundingAgent.feature('regexp-match-indices') ? 'dgimsuy' : 'gimsuy').includes(c)
```
Comment (devsnek) [inline src/parser/Lexer.mjs:891]: you can use `this.feature` instead.

Comment (rbuckton) [reply src/parser/Lexer.mjs]: If only there were a way to indicate `Lexer` had an abstract method named `feature` that must eventually be implemented by a subclass like `Parser`... 😉

**Comment by rbuckton** [comment]
tc39/test262#2934 has been merged

**Comment by devsnek** [comment]
Can you just update the test262 submodule so this is green?

**Pull request merged**